File: backend/src/main/java/com/inventory/backend/repository/RoomRepository.java
package com.inventory.backend.repository;

import com.inventory.backend.entity.Room;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;

public interface RoomRepository extends JpaRepository<Room, Long> {
    List<Room> findAllByOrderByNameAsc();
    List<Room> findByDepartmentIdOrderByNameAsc(Long departmentId);
    long countByDepartmentId(Long departmentId);
    boolean existsByDepartmentIdAndNameIgnoreCase(Long departmentId, String name);
    boolean existsByDepartmentIdAndNameIgnoreCaseAndIdNot(Long departmentId, String name, Long id);
}

File: backend/src/main/kotlin/com/inventory/backend/repository/RoomRepository.kt
package com.inventory.backend.repository

import com.inventory.backend.entity.Room
import com.inventory.backend.entity.RoomStatus
import org.springframework.data.jpa.repository.JpaRepository

interface RoomRepository : JpaRepository<Room, Long> {
    fun findAllByOrderByNameAsc(): List<Room>
    fun findByDepartmentIdOrderByNameAsc(departmentId: Long): List<Room>
    fun countByDepartmentId(departmentId: Long): Long
    fun existsByDepartmentIdAndStatus(departmentId: Long, status: RoomStatus): Boolean
    fun existsByDepartmentIdAndNameIgnoreCase(departmentId: Long, name: String): Boolean
    fun existsByDepartmentIdAndNameIgnoreCaseAndIdNot(departmentId: Long, name: String, id: Long): Boolean
}
